push.bash

```diff
@@ -33,4 +41,23 @@ if [ -z "$BUF_COMMAND" ]; then
   fail "$NOT_INSTALLED_MESSAGE"
 fi
 
-BUF_TOKEN="${BUF_TOKEN}" "${BUF_COMMAND}" push --tag "${GITHUB_SHA}" "${BUF_INPUT}"
+if [ "${GITHUB_REF_TYPE}" != "branch" ]; then
+  echo "reference type is not branch, skipping" >&2
+  exit 0
+fi
+
+BUF_ARGS=("--tag" "${GITHUB_SHA}")
+if [ "${GITHUB_REF_NAME}" != "main" ] && [ "${ENABLE_DRAFT}" == "true" ]; then
```

Comment:
Not everyone's main branch will be named main, there's probably still people using the old default `master`. Like this I think it would be a breaking change, I think we need to check if we can get the main branch name somehow.

From the previous docs and script it seems like we currently expect people to have this setup as
```
on: # Apply to all pushes to `main`
  push:
    branches:
      - main
jobs:
  validate-and-push-protos:
    runs-on: ubuntu-latest
    steps:
...
      - uses: bufbuild/buf-push-action@v1
        with:
          buf_token: ${{ secrets.BUF_TOKEN }}
```
Before, we would have pushed any branch to bsr as non draft:
```
on:
  push:
    branches:
      - master
jobs:
...
      - uses: bufbuild/buf-push-action@v1
        with:
          buf_token: ${{ secrets.BUF_TOKEN }}
```
If users already need to configure it, maybe makes sense if `draft: true` always pushes to a draft regardless of branch name? (even if it's main, in which case I think we'd error).

Reply:
> This will mean that the user will need to invoke the buf-push-action at least twice it they want a complete flow of pushing draft on PR and push to main on merge

They'll at least need to remove the `main` from the on branches push or add a `on: pr`? I would expect github actions's expressions allows us to do something like:
```diff
 on:
   push:
-    branches:
-    - master
 jobs:
   ...
       - uses: bufbuild/buf-push-action@v1
         with:
           buf_token: ${{ secrets.BUF_TOKEN }}
+          draft: ${{ env.GITHUB_REF == "main"}}
```

Reply:
yeh this will work, the only thing remaining is that I am not sure about the UX here, this will put the logic into user's workflow config, where we actually able to include it as part of our action, which might be a bit verbose / not out of the box to the user?

Reply:
If the alternative is requiring users to have a param with the name of their main branch, I feel like this is more straightforward (at least if we have an example).

Reply:
sounds reasonable! updated, also tested in my repo that the example we provide will work